### PR TITLE
Implement essential claims support to OAuth2

### PR DESCRIPTION
**Pull request by Nilki**
### Proposed changes in this pull request

Implement the code for supporting essential claims and essential claims with value.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -796,6 +796,9 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-checkstyle-plugin</artifactId>
+                <configuration>
```
Comment (senthalan) [inline pom.xml:799]: We can't skip checkstyle plugins

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java [hunk 2]
```diff
@@ -45,20 +47,12 @@
...
-import java.util.Arrays;
-import java.util.Calendar;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java:55]: We don't use * imports

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 3]
```diff
@@ -3085,7 +3085,89 @@ private static String hexify(byte bytes[]) {
         }
         return essentialClaimsList;
     }
+   /**private static Map<String, String> filterEssentialClaimJson(String essentialClaims, String claimType){
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:3088]: Why do we have this code block commented ?

**Review by senthalan — CHANGES_REQUESTED** [review]

**Comment by Nilki** [comment]
Please refer new PR [https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1789]

**Pull request closed without merging**